drivers/dome/scopedome_dome.cpp

```diff
@@ -242,383 +209,423 @@ bool ScopeDome::SetupParms()
         SetAxis1ParkDefault(0);
     }
 
-    uint8_t calibrationNeeded = false;
-    readU8(IsFullSystemCalReq, calibrationNeeded);
-    CalibrationNeededS[0].s = calibrationNeeded ? ISS_ON : ISS_OFF;
-    CalibrationNeededSP.s   = IPS_OK;
-    IDSetSwitch(&CalibrationNeededSP, nullptr);
-
-    uint16_t fwVersion;
-    readU16(GetVersionFirmware, fwVersion);
-    FirmwareVersionsN[0].value = fwVersion / 100.0;
-
-    uint8_t fwVersionRotary;
-    readU8(GetVersionFirmwareRotary, fwVersionRotary);
-    FirmwareVersionsN[1].value = (fwVersionRotary + 9) / 10.0;
-    FirmwareVersionsNP.s       = IPS_OK;
-    IDSetNumber(&FirmwareVersionsNP, nullptr);
+    bool calibrationNeeded = interface->isCalibrationNeeded();
+    CalibrationNeededSP[0].setState(calibrationNeeded ? ISS_ON : ISS_OFF);
+    CalibrationNeededSP.setState(IPS_OK);
+    CalibrationNeededSP.apply();
+
+    double fwMain, fwRotary;
+    interface->getFirmwareVersions(fwMain, fwRotary);
+    FirmwareVersionsNP[0].setValue(fwMain);
+    FirmwareVersionsNP[1].setValue(fwRotary);
+    FirmwareVersionsNP.setState(IPS_OK);
+    FirmwareVersionsNP.apply();
+
+    rotaryLinkEstablished = interface->getInputState(ScopeDomeCard::ROTARY_LINK);
+    if(rotaryLinkEstablished == ISS_OFF)
+    {
+        LOG_WARN("Rotary link not established, shutter control not possible");
+    }
     return true;
 }
 
 /************************************************************************************
  *
-* ***********************************************************************************/
+ ************************************************************************************/
 bool ScopeDome::Handshake()
 {
-    return Ack();
+    if(reconnecting)
+        return true;
+
+    sim = isSimulation();
+
+    if (sim)
+    {
+        interface.reset(static_cast<ScopeDomeCard *>(new ScopeDomeSim()));
+    }
+    else
+    {
+        if(CardTypeSP[0].getState() == ISS_ON)
```

Comment:
Better use enums instead of CardTypeSP[0]. i.e. CardTypeSP[CARD_USB] ...etc

Reply:
True and with FindOnSwitchIndex much cleaner